src/core/annotation.js

```diff
@@ -485,7 +485,7 @@ var LinkAnnotation = (function LinkAnnotationClosure() {
     data.annotationType = AnnotationType.LINK;
 
     var action = dict.get('A');
-    if (action) {
+    if (isDict(action)) {
```

Comment:
How about `if (action && isDict(action)) {` instead, to avoid doing the `isDict` check if `action` isn't defined?